modules/kargoBidAdapter.js

```diff
@@ -4,286 +4,481 @@ import { registerBidder } from '../src/adapters/bidderFactory.js';
+  const validPageUrl = spec._getLocalStorageSafely(CERBERUS.PAGE_VIEW_URL) != null
+
+  const page = {}
+  if(validPageId) {
+    page.id = spec._getLocalStorageSafely(CERBERUS.PAGE_VIEW_ID)
+  }
+  if(validPageTimestamp) {
+    page.timestamp = Number(spec._getLocalStorageSafely(CERBERUS.PAGE_VIEW_TIMESTAMP))
+  }
+  if(validPageUrl) {
+    page.url = spec._getLocalStorageSafely(CERBERUS.PAGE_VIEW_URL)
+  }
+  if (!_.isEmpty(page)) {
+    krakenParams.page = page
+  }
+  
+  return Object.assign({}, bidderRequest, {
+    method: BIDDER.REQUEST_METHOD,
+    url: `${BIDDER.HOST}${BIDDER.REQUEST_ENDPOINT}`,
+    data: krakenParams,
+    currency: currency
+  });
+}
+
+function interpretResponse(response, bidRequest) {
+  let bids = response.body;
+  const bidResponses = [];
+  for (let bidId in bids) {
+    let adUnit = bids[bidId];
+    let meta = {
+      mediaType: BANNER
+    };
+
+    if (adUnit.metadata && adUnit.metadata.landingPageDomain) {
+      meta.clickUrl = adUnit.metadata.landingPageDomain[0];
+      meta.advertiserDomains = adUnit.metadata.landingPageDomain;
     }
 
-    return bidResponses;
-  },
-  getUserSyncs: function(syncOptions, responses, gdprConsent, usPrivacy) {
-    const syncs = [];
-    const seed = spec._generateRandomUuid();
-    const clientId = spec._getClientId();
-    var gdpr = (gdprConsent && gdprConsent.gdprApplies) ? 1 : 0;
-    var gdprConsentString = (gdprConsent && gdprConsent.consentString) ? gdprConsent.consentString : '';
-    // don't sync if opted out via usPrivacy
-    if (typeof usPrivacy == 'string' && usPrivacy.length == 4 && usPrivacy[0] == 1 && usPrivacy[2] == 'Y') {
-      return syncs;
+    if (adUnit.mediaType && BIDDER.SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType)) {
+      meta.mediaType = adUnit.mediaType;
     }
-    if (syncOptions.iframeEnabled && seed && clientId) {
-      for (let i = 0; i < SYNC_COUNT; i++) {
-        syncs.push({
-          type: 'iframe',
-          url: SYNC.replace('{UUID}', clientId).replace('{SEED}', seed)
-            .replace('{INDEX}', i)
-            .replace('{GDPR}', gdpr)
-            .replace('{GDPR_CONSENT}', gdprConsentString)
-            .replace('{US_PRIVACY}', usPrivacy || '')
-        });
-      }
+
+    const bidResponse = {
+      ad: adUnit.adm,
+      requestId: bidId,
+      cpm: Number(adUnit.cpm),
+      width: adUnit.width,
+      height: adUnit.height,
+      ttl: 300,
+      creativeId: adUnit.id,
+      dealId: adUnit.targetingCustom,
+      netRevenue: true,
+      currency: adUnit.currency || bidRequest.currency,
+      mediaType: meta.mediaType,
+      meta: meta
+    };
+
+    if (meta.mediaType == VIDEO) {
+      bidResponse.vastXml = adUnit.adm;
     }
+
+    bidResponses.push(bidResponse);
+  }
+
+  return bidResponses;
+}
+
+function getUserSyncs(syncOptions, responses, gdprConsent, usPrivacy) {
+  const syncs = [];
+  const seed = spec._generateRandomUuid();
```

Comment:
I think these can't reference `spec` anymore since it's now a function on the file itself and not part of the `spec` object
```suggestion
  const seed = _generateRandomUuid();
```

Reply:
I have removed this reference where I can. Some need to be maintain it to ensure unit test stubbing works correctly